homeassistant/components/fibaro/__init__.py

```diff
@@ -355,7 +399,8 @@ def _read_devices(self):
                     device.mapped_platform = None
                 if (platform := device.mapped_platform) is None:
                     continue
-                device.unique_id_str = f"{self.hub_serial}.{device.id}"
+                device.unique_id_str = f"{slugify(self.hub_serial)}.{device.id}"
+                self._create_device_info(device, devices)
```

Comment:
Why is this called here and not when the device info is requested ?

Reply:
The device itself does not contain enough information to create device info, therefore I create it upfront in the controller where all information is available. So if we like to change that I think we need to first do an intensive refactoring of the FibaroController and device classes. So if you think the refactoring should be done first just mention it here, then I will try to refactor those classes upfront in a separate PR and decline this one.